### DISCUSSION TITLE
How to debug disappearing inserts in a cluster?

### DISCUSSION THREAD

**Discussion by fagzal** [ClickHouse/ClickHouse#43241] [Q&A] [answered]
So I have 3 systems,  a single CH server, a 2 node test cluster, and a 3 node productive cluster.
I am loading 5 TSV files into them - not very big ones, about 200k lines, 140 columns each.
(It's the same set of files for testing, so it's not a data dependent issue.)

The load always work in the single instance, but some of the files (always different ones...) fail *sometimes* on both the test and prod cluster. It's totally random. However, there is no error message whatsoever: whether I'm using a driver (HTTP, from PHP) or the clickhouse-client CLI, there is no error message (the server nodes doesn't have any error messages in the logs, either), the client shows no error... but the data is not in the table.

In the single server, the table is MergeTree, on the nodes, they are ReplicatedMergeTree.

Any ideas how can I debug where the inserts disappear, other than the logs??
----
Related, I also have one specific TSV file that is 40k lines, that fails to load every time. But if I spit it into 2 files (20-20k lines) it loads without errors.
For the 40k lines, the error message is:

Code: 62. DB::Exception: Syntax error: failed at position 321 ('some_field') (line 4, col 264): some_field = 'AIR' FORMAT JSON. Expected one of: token, DoubleColon, MOD, DIV, NOT, BETWEEN, LIKE, ILIKE, NOT LIKE, NOT ILIKE, IN, NOT IN, GLOBAL IN, GLOBAL NOT IN, IS, AND, OR, QuestionMark, alias, AS, GROUP BY, WITH, HAVING, WINDOW, ORDER BY, LIMIT, OFFSET, SETTINGS, UNION, EXCEPT, INTERSECT, INTO OUTFILE, FORMAT, end of query.

Loading with simply : cat file.tsv | clickhouse-client --query="INSERT INTO mytable FORMAT TabSeparated"

(Server version 22.8.4 revision 54460)

**Comment by den-crane** [comment]
1. https://kb.altinity.com/altinity-kb-schema-design/insert_deduplication/

**Reply by fagzal** [reply]
Well, not this one, as I truncate the table between the testing runs. Sorry I did not specify that!

**Reply by fagzal** [reply]
Oh okay, maybe you are right here...

**Comment by den-crane** [comment, marked as answer]
2. it's a bug, probably fixed, 22.8.4 is out support, you have to upgrade 22.8.8.3. 
try `cat file.tsv | clickhouse-client --input_format_parallel_parsing=0 --query="INSERT INTO mytable FORMAT TabSeparated"`

**Reply by fagzal** [reply]
Well, probably this one :) Experimenting with 22.10 now, and the issue no longer reproduces...
Thank you, I will post it here one I'm 100% sure.
